### PR TITLE
FF7: Implemented GPU-based original game lighting that better matches PSX version

### PR DESCRIPTION
**Pull request by CosmosXIII**
## Summary

I re-implemented the original lighting done by the game in the shaders instead of the original CPU-based implementation.
In doing so I found that the light directions used by the game were inverted compared to the PSX version, so the new modes fix that too. The new modes better match better the PSX version lighting but there are small differences in contrast. Part of this is due to FFNx normal auto-calculation so those may not fully match.

### Motivation

To make people happy. Also, if we end up doing smooth skinning, doing lighting on the GPU is necessary so this paves the path for that.

### ACKs

- [x] I have updated the [Changelog.md](https://github.com/julianxhokaxhiu/FFNx/blob/master/Changelog.md) file
- [x] I did test my code on FF7
- [ ] I did test my code on FF8

### CODE REVIEW

File: src/renderer.cpp [hunk 1]
```diff
@@ -394,6 +411,9 @@ void Renderer::resetState()
     setOverallColorGamut(enable_ntscj_gamut_mode ? COLORGAMUT_NTSCJ : COLORGAMUT_SRGB);
     setGamutOverride();
     setGammaType();
+	setGameLightData();
```
Comment (julianxhokaxhiu) [inline src/renderer.cpp:414]: Can you please fix the indentation?

File: misc/FFNx.lighting.frag [hunk 2]
```diff
@@ -104,9 +113,12 @@ uniform vec4 iblData;
 #define isPbrTextureLoaded FSTexFlags.y > 0.0
 #define isIblTextureLoaded FSTexFlags.z > 0.0
 
+#define gameLightingMode gameLightingFlags.x
+#define GAME_LIGHTING_PER_PIXEL 2
+
 void main()
 {
-	vec4 color = vec4(toLinear(v_color0.rgb), v_color0.a);
+	vec4 color = v_color0;
```
Comment (julianxhokaxhiu) [inline misc/FFNx.lighting.frag:121]: Can you please fix the indentation here to match the lines below?

**Review by julianxhokaxhiu — CHANGES_REQUESTED** [review]
Minor indentation issues, but the PR looks good to me. Thanks, looking forward to it :)

File: src/renderer.cpp (hunk 1 above)
Comment (CosmosXIII) [reply src/renderer.cpp]: Fixed :)

File: misc/FFNx.lighting.frag (hunk 2 above)
Comment (CosmosXIII) [reply misc/FFNx.lighting.frag]: Fixed :)

**Comment by julianxhokaxhiu** [comment]
Thank you :)

**Pull request merged**